app/src/main/java/me/hufman/androidautoidrive/phoneui/viewmodels/PermissionsModel.kt

```diff
@@ -80,6 +87,8 @@ class PermissionsModel(private val notificationListenerState: LiveData<Boolean>,
 			}
 			else -> _spotifyErrorHint.value = { errorMessage ?: "" }
 		}
+
+		_isSpotifyWebApiAuthorized.value = spotifyAuthStateManager.isAuthorized()
```

Comment:
This can be moved to the main `fun updateSpotify()` to restore the state immediately. This `fun _updateSpotify()` is called after the SpotifyAppController finishes its attempt to connect to the Spotify App, and adds a slight delay.

Reply:
Moved this logic to the higher level updateSpotify() method

Reply:
Oh actually it should be in both places, the copy in _update is to reflect any auth changes after the SpotifyWebApi is created 😅

Reply:
Oh got it, I added it back to the _updateSpotify() so now it's in both places